sql/table.go

```diff
@@ -202,6 +205,13 @@ func encodeTableKey(b []byte, val parser.Datum) ([]byte, error) {
 		return encoding.EncodeFloat(b, float64(t)), nil
 	case parser.DString:
 		return encoding.EncodeString(b, string(t)), nil
+	case parser.DDate:
+		sec := t.Time.Unix()
+		return encoding.EncodeVarint(b, sec), nil
+	case parser.DTimestamp:
+		return encoding.EncodeTime(b, t.Time), nil
```

Comment:
I would have thought the `DDate` and `DTimestamp` encodings would be the same given they are both backed by `time.Time`. If there is a good reason for the difference it should be documented in a comment.